### PR TITLE
Introduce backoff for failing jobs

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

File: test/spec/job/processor.js [hunk 1]
```diff
@@ -97,15 +97,15 @@ describe('JobProcessor', function() {
...
+    var processor = new JobProcessor(tempJobsDir, null, 1);
     processor.start();
     var jobPromise1 = processor.process(job1);
     var jobPromise2 = processor.process(job2);
     Promise.all([jobPromise1, jobPromise2])
       .timeout(300 + 50)
       .catch(Promise.TimeoutError, function() {
         assert.isTrue(jobPromise1.isFulfilled());
-        assert.isTrue(jobPromise2.isCancelled());
+        assert.isTrue(jobPromise2.isPending());
```
Comment (tomaszdurka) [inline test/spec/job/processor.js]: Why this change? :)

**Comment by tomaszdurka** [comment]
Looks good, tests are failing.

File: test/spec/job/processor.js (hunk 1 above)
Comment (vogdb) [inline test/spec/job/processor.js]: well, this test is incorrect

Comment (vogdb) [inline test/spec/job/processor.js]: hm, actually correct. Why it failed on my machine >_<

**Comment by vogdb** [comment]
@tomaszdurka please look

**Comment by tomaszdurka** [comment]
lgtm
